### PR TITLE
github-actions: use new env API

### PR DESCRIPTION
**Pull request by alltilla**
Based upon #3484. I have opened a new PR, because this problem just become critical, as `set-env` is removed, and nearly all of our CI jobs fail.

Signed-off-by: Attila Szakacs <attila.szakacs@oneidentity.com>

### CODE REVIEW

**Comment by alltilla** [comment]
distcheck fails:
```make[4]: *** No rule to make target 'modules/python/pylib/test_pylib.sh'.  Stop.```
Is this a new problem?

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: .github/workflows/devshell.yml [hunk 1]
```diff
@@ -144,7 +144,7 @@ jobs:
         run: ./autogen.sh
 
       - name: configure
-        run: ./configure --disable-all-modules
+        run: ./configure ${DISTCHECK_CONFIGURE_FLAGS}
```
Comment (gaborznagy) [inline .github/workflows/devshell.yml]: This patch actually is needed. We discussed this with @bazsi that we should test `make distcheck` with a tarball that is built in a minimal environment.
`make distcheck` will use a different set of configure flags through `DISTCHECK_CONFIGURE_FLAGS`
He even found an issue around Python, IIRC.
We can discuss this IRL.

**Comment by gaborznagy** [comment]
> distcheck fails:
> `make[4]: *** No rule to make target 'modules/python/pylib/test_pylib.sh'. Stop.`
> Is this a new problem?

I guess that's what @bazsi found when there is no python when the tarball is created: https://github.com/syslog-ng/syslog-ng/pull/3484/commits/07c4103a0292b7ba7cab70ce60d83443a1854075

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: .github/workflows/devshell.yml (hunk 1 above)
Comment (alltilla) [reply .github/workflows/devshell.yml]: Oh, I see. That is another kind if change, tho, not related to the github actions env API.

Comment (bazsi) [reply .github/workflows/devshell.yml]: yup. I've just found a few other cases. These might go into a separate PR though.

**Review by gaborznagy — DISMISSED** [review]

**Review by gaborznagy — DISMISSED** [review]

**Comment by alltilla** [comment]
I moved the `gh_export()` function to a separate file, which just needs to be sourced. If the CI jobs became green, I would consider the PR final.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Review by gaborznagy — APPROVED** [review]

**Review by lbudai — APPROVED** [review]

**Comment by bazsi** [comment]
I like the way you could share code with the steps (e.g. gh_export.sh). I was looking for a means to do this, but couldn't find a solution. Thanks.

**Pull request merged**